WooCommerce/WooCommerceTests/POS/ViewModels/ItemListViewModelTests.swift

```diff
@@ -352,37 +417,14 @@ final class ItemListViewModelTests: XCTestCase {
 
         XCTAssertTrue(sut.showSimpleProductsModal)
     }
-
-    func test_currentPage_when_populatePointOfSaleItems_is_invoked_multiple_times_then_updates_to_next_page() async {
-        XCTAssertTrue(sut.currentPage == 0, "Initial state.")
-
-        // When/Then
-        await sut.populatePointOfSaleItems()
-        XCTAssertTrue(sut.currentPage == 1)
-
-        // When/Then
-        await sut.populatePointOfSaleItems()
-        XCTAssertTrue(sut.currentPage == 2)
-    }
-
-    func test_currentPage_when_reload_is_invoked_multiple_times_then_stays_on_first_page() async {
-        XCTAssertTrue(sut.currentPage == 0, "Initial state.")
-
-        // When/Then
-        await sut.reload()
-        XCTAssertTrue(sut.currentPage == 1)
-
-        // When/Then
-        await sut.reload()
-        XCTAssertTrue(sut.currentPage == 1)
-    }
 }
 
 private extension ItemListViewModelTests {
     final class MockPOSItemProvider: POSItemProvider {
         var items: [POSItem] = []
         var shouldThrowError = false
         var shouldReturnZeroItems = false
+        var shouldSimulateFetchNextPage = false
```

Comment:
nit: alternatively, `items` can be the only way to return the items in the test cases (so that `items` is more like "items to return in `providePointOfSaleItems` regardless of the page"), and this property `shouldSimulateFetchNextPage` isn't needed. `simulateFetchNextPage` can return a list of items different from `makeInitialItems`.

Reply:
Oh that's true 🤔 I'm going to give another pass to unit tests soon, along with the comment on https://github.com/woocommerce/woocommerce-ios/pull/14211#pullrequestreview-2394482146

> _I think adding a property like pageNumber to MockPOSItemProvider (only in unit tests) would be feasible without having to expose any state in the app code._